File: Data Structures & Algorithms/majority-element/submission-1.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        
        int a = nums.size() / 2;
        std::unordered_map<int, int> myMap;

        for (int i = 0; i < nums.size(); i++) {
            if (myMap[nums[i]] == a) {
                return nums[i];
            }
            myMap[nums[i]]++;
        }
    }
};

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-1.py
from collections import defaultdict

class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        ps = defaultdict(int)
        ps[0] = 1
        res = 0
        for i in range(1, len(nums) + 1):
            currSum = sum(nums[0:i])
            diff = currSum - k
            if diff in ps:
                res += ps[diff]
            ps[currSum] += 1
        
        return res
            